src/analyzer/plugins/analyzerqueenmarykey.cpp

```diff
@@ -86,9 +86,7 @@ bool AnalyzerQueenMaryKey::initialize(int samplerate) {
 }
 
 bool AnalyzerQueenMaryKey::processSamples(const CSAMPLE* pIn, const int iLen) {
-    DEBUG_ASSERT(iLen == kAnalysisSamplesPerBlock);
```

Comment:
Why do we pad the first block with extra zeros, but not the final block even it is only partially filled with sample data? This is inconsistent and difficult to understand.